ethstorage/es-node
# Sync perf -- Run nodes with network=perf to start performance test on multi machines. (draft)

Run nodes with network=perf to start performance test on multi machines:

How to test:
1. create a storage file
`./es-util create --filename storage.dat --chunk_size=131072 --kv_size=131072 --kv_len=4096 --miner=0x5C935469C5592Aeeac3372e922d9bCEabDF8830d`
2. fill in the storage file with random data, and create a related meta file for mock l1 
`./es-util fill_up_shard --filename storage.dat --shard_idx=0 --kv_entries=256 --kv_size 131072 --chunk_size 131072 --metafile storage.dat.meta`

3. start nodes with `network=perf`, and `--l1.mock_metafile storage.dat.meta` on the local machine as a sync server to provide blobs.
`--network perf --p2p.listen.ip 127.0.0.1 --p2p.listen.udp 30303 --p2p.listen.tcp 9222 --p2p.advertise.ip 127.0.0.1  --l1.rpc http://65.108.236.27:8545 --l1.beacon http://65.109.63.154:5052 --storage.files storage.dat --storage.l1contract 0x8FA01287747B8f876C8bA532D9Bd48b73329c431 --storage.miner 0x5C935469C5592Aeeac3372e922d9bCEabDF8830d --storage.kv-size 131072  --storage.chunk-size 131072 --storage.kv-entries 256 --datadir ./database --download.start -3 --log.level debug  --l1.mock_metafile storage.dat.meta`
4. create an empty storage file, copy the local metafile to remote machines, and start the node with perf network and l1.mock_metafile as sync client to sync blobs.
note: to test runtime metrics on sync clients or servers, sync client and server can be multiple.

how to verify:
after the test start, log for runtime metrics will print like the following
`INFO [10-09|19:34:20.257] runtime metrics                          "cpu (%)"=55 "memory (MB)"=9198 "disk read (KB)"=69727 "disk write (KB)"=144,879`
